server/room.controller.js

```diff
@@ -0,0 +1,63 @@
+const Room = require('./room.model.js');
+
+let waitingRooms = [];
+
+//Room functions
+
+const onPlayerWaiting = function(playerId, topicId) {
+  const roomIndex = waitingRooms.findIndex((r) => {
+    return r.topicId === topicId;
+  });
+  if(roomIndex !== -1) {
+    updateRoom(waitingRooms[roomIndex], playerId);
+  } else {
+    createRoom(topicId, playerId);
+  }
+}
+
+const createRoom = function(topicId, playerId) {
+  //socket.broadcast.emit('broadcast(): player ' + playerId + ' joined room with topic' + topicId);
+
+  const room = new Room(topicId, onRoomTimeOut, onRoomFull);
+  room.addPlayer(playerId);
+  waitingRooms.push(room);
+  //socket.emit("playerConnected", {playerId, topicId});    
+};
+
+const updateRoom = function(room, playerId) {
+  if (room.isAvailable) {
+    //socket.broadcast.emit('broadcast(): player ' + playerId + ' joined room with topic' + topicId);
+    room.addPlayer(playerId);
+    //socket.emit("playerConnected", {playerId, topicId});
+  }
+};
+
+const removeRoom = function(topicId) {
+  const roomIndex = waitingRooms.findIndex((r) => {
+    return r.topicId === topicId;
```

Comment:
r.getRoom() ?? And isnt the property name _topic?

Reply:
there is a get topicId in room.js